### DISCUSSION TITLE
SSR을 위한 UI 라이브러리 변경

### DISCUSSION THREAD

**Discussion by minjongbaek** [prgrms-web-devcourse/Team-Gaerval-Dadok-FE#315] [General]
SSR을 적극 활용하자는 의견을 잠깐 나눴던 것 같아요!

다독다독에서 사용하고 있는 Chakra UI를 포함하여 CSS in JS 방식으로 구현된 대부분의 UI 라이브러리가 SSR을 지원하지 않는 것으로 알고 있어요.

몇 자료를 찾아본 결과 `Tailwind CSS`나 `Styled-Component` 로 서버 컴포넌트를 구현할 수 있는 것 같아요.

어떤 라이브러리를 사용하는게 우리에게 더 좋을지 의견을 나누면 좋을 것 같아요.

**Comment by harryk-im** [comment]
CSR과 SSR을 적절히 섞어서 페이지 별로 역할을 나누어 이를 구현해 본다면 분명 팀원들에게 좋은 경험이 될 것이라고 생각합니다

개인적으로 다른 라이브러리를 채택하는 부분에서는 이렇게 생각하고있어요.

- 다들 이력서와 포트폴리오를 준비하고 있는 기간, 아무래도 할애되는 시간이 이전보다는 짧을것이라고 생각
- 사용자가 꾸준히 많고 안정적인 라이브러리를 사용하자. (교체중 일어나는 트러블 슈팅이 비교적 수월할 듯)
- 러닝커브가 낮고 팀원들이 한번 쯤 사용해본 라이브러리를 사용하자.

이러한 생각들을 바탕으로 저는 중간프로젝트에 다들 한번씩 사용해본
`styled-component` (혹은 `emotion`)을 사용해보면 좋을 것 같다 생각합니다.


자세한 이야기는 회의일정을 잡고 함께 의견을 나누어 보면 좋을것 같습니다.

**Comment by harryk-im** [comment]
아무래도 CSS 라이브러리를 싹다 갈아치우는게 진짜 쉽지않은 일이라서 방법을 좀 생각해보다가

SSG의 단점을 어느정도 보완한 ISR을 사용해보는건 어떨까하는 생각도 드네요.

먼저 ISR을 적용해보고 퀄리티가 기대치 이하라면 CSS 라이브러리를 갈아치우는 선택을 하는게 좋을것 같아요.

**Reply by minjongbaek** [reply]
isr을 적용하면 콘텐츠가 변경된 후에 사이트를 다시 방문하게 되어도 이전의 콘텐츠를 보게 될 수 있다는 단점이 있을 것 같아요! 초기 렌더링에 변화가 없는 검색 페이지와 같은 예는 적합할 수 있겠네요!
